lib/mail/notify/mailer.rb

```diff
@@ -84,9 +84,28 @@ def view_mail(template_id, options)
 
         # we have to render the view for the message and grab the raw source,
         # then we set that as the body in the personalisation for sending to
-        # the Notify API.
+        # the Notify API. 
+        #
+        # Calling the #mail method is not idempotent. It
+        # modifies state by setting instance variables on the message.
+        # Specifically, it sets @_message. mail generates message headers for
+        # the options passed in. Each time it is called with the same headers
+        # it adds another header field. This results in something like this
+        #
+        # mail({custom_header => 123})
+        # message.header['custom_header']
+        # #=> Mail::Field
+        #
+        # mail({custom_header => 123})
+        # message.header['custom_header']
+        # #=> [Mail::Field..., Mail::Field...]
+        #
+        original_message = message.dup
+
         body = mail(headers).body.raw_source
```

Comment:
I think we can just not pass `headers` to the call to `mail` here and leave it to the final call that will be returned.

Reply:
Digging into it more, I can't see why we are handling headers other than `to` and `subject` at all, they will never be used as we cannot pass them onto Notify via the API.

I would be interested to know what headers are being duplicated for you and are showing up in your tests.

I've refactored the two methods to only use to and subject on the end of #165.